File: DS-n-Algos-Mega-Archive/JAVASCRIPT/Data.Structures.and.Algorithm/JavaScript/Stacks/Stacks.js
class Stack {
	constructor() {
		this.elements = []
		this.top = -1
	}

	push(item) {
		this.elements[++this.top] = item
	}

	pop() {
		if (top == -1)
			return 'Stack Underflow'

		return this.elements[this.top--]
	}

	printStack() {
		const { elements, top } = this
		for (let i = 0; i <= top; i++) {
			console.log(elements[i])
		}
	}
}


let s = new Stack()
s.push(5)
s.push(10)
s.printStack() // prints 5, 10

s.pop() // returns 10
s.printStack() // prints 5

File: DS-n-Algos-Mega-Archive/JAVASCRIPT/Data.Structures.and.Algorithm/Kotlin/src/stacks/Stacks.kt
package stacks

class Stack {
    var arr = ArrayList<Int>()
    var topIndex = -1

    fun push(item: Int) {
        ++topIndex
        arr.add(topIndex, item)
    }

    fun pop(): Int = arr.removeAt(topIndex--)

    fun displayStack() {
        println("Stacks.Stack elements are: ")
        for (i in 0..topIndex) {
            println(arr[i])
        }
    }
}

fun main() {
    val stack = Stack()
    stack.push(5)
    stack.push(10)
    stack.push(9)
    stack.displayStack()
    val poppedItem = stack.pop()
    println("Popped element $poppedItem")
    stack.displayStack()
}